### PR TITLE
Forcing the removal of corrupt 3rd-party repos

### LINKED ISSUE #1343: Make swupd 3rd-party remove more error prone
Right now swupd 3rd-party remove depens on the installed 3rd-part content to be sane in order to proceed. This is problematic because if there's a corruption (or destructive changes) in the installed directory there's no way to remove that permanently.

For example the following would fail:
```
# sudo ./swupd 3rd-party add repo http://my_mix
# rm /opt/3rd-party/bundles/repo/usr/lib/os-release
# sudo ./swupd 3rd-party remove repo
```

We need a better approach on when there are errors removing the wrappers. Maybe a --force to remove the repo anyway? And for sure better warning messages.

### PR DESCRIPTION
**Pull request by castulo**
This commit allow users to force the deletion of corrupt 3rd-party repos
which will end up deleting most of the repo's files, except for the
scripts that exported files from that repo.

Closes #1343

Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

### CODE REVIEW

**Comment by castulo** [comment]
@otaviobp please check the messages I'm printing and let me know if we need to provide more information to the user.

File: src/3rd_party_remove.c [hunk 1]
```diff
@@ -146,10 +167,29 @@ enum swupd_code third_party_remove_main(int argc, char **argv)
...
 
+	/* if there are errors at this point the repo may be corrupt,
+	 * continue only if forced by the user */
+remove_repo:
+	if (ret) {
+		if (cmdline_option_force) {
+			char *bin_dir;
+			bin_dir = third_party_get_bin_dir();
+			warn("The --force option is specified; forcing the removal of the repository\n");
+			info("Please be aware that this may leave dangling files in %s\n", bin_dir);
```
Comment (otaviobp) [inline src/3rd_party_remove.c]: s/dangling/exported/ ?

File: src/3rd_party_remove.c [hunk 2]
```diff
@@ -146,10 +167,29 @@ enum swupd_code third_party_remove_main(int argc, char **argv)
...
+			free_and_clear_pointer(&bin_dir);
+			ret_partial = ret;
+			ret = SWUPD_OK;
+		} else {
+			error("The 3rd-party repository %s is corrupt and cannot be removed cleanly\n", name);
```
Comment (otaviobp) [inline src/3rd_party_remove.c]: I think it's better to write something here about the possible dangling files too. Something like:
```
The 3rd-party repository %s is corrupt and cannot be removed cleanly
Some exported files can be left behind at /opt/3rd-party/bin/
```

Comment (castulo) [reply src/3rd_party_remove.c]: sounds good, I'll make the change

File: src/3rd_party_remove.c (hunk 1 above)
Comment (castulo) [reply src/3rd_party_remove.c]: sure, I'll go with 'exported' then

**Pull request merged**